pageserver/src/waldecoder.rs

```diff
@@ -738,7 +740,15 @@ pub fn decode_wal_record(record: Bytes) -> DecodedWALRecord {
                     let spcnode = buf.get_u32_le();
                     let dbnode = buf.get_u32_le();
                     let relnode = buf.get_u32_le();
-                    //TODO handle this too?
+                    let mut blk = DecodedBkpBlock::new();
+                    blk.tag = ObjectTag::RelationMetadata(RelTag {
+                        forknum: pg_constants::MAIN_FORKNUM,
+                        spcnode,
+                        dbnode,
+                        relnode,
+                    });
+                    blk.will_drop = true;
+                    blocks.push(blk);
```

Comment:
Seems confusing to store a RelationMetadata key in a variable called `blk.tag`. Because it's clearly not a block. This needs some more refactoring to make sense.

It seems we're abusing the DecodedBkpBlock beyond what it was intended to represent. For comparison, truncation records are handled in `save_decoded_record`, I think that would be the right place to handle relation drops too.

Hmm, before this commit, is this code for parsing XACT records actually used for anything? I think I missed this in the earlier revert of non-rel file handling (commit 6a9c036ac165de37fcb449c467ed2a2985d37511).

Reply:
Created a PR to clean that up: https://github.com/zenithdb/zenith/pull/272. Please take a look